graphistry/umap_utils.py

```diff
@@ -361,7 +434,10 @@ def umap(
         :return: self, with attributes set with new data
         """
         assert_imported()
+        assert_imported_cuml()
         self.umap_lazy_init()
+        engine_resolved = resolve_umap_engine(engine)
+        
 
         self.suffix = suffix
```

Comment:
all assignments should be on `self`, and after `if inplace: res = self else res = self.bind()`...

Reply:
ok i missed that one point and have changed it in the wimpiest commit ever